### PR TITLE
Small warnings cleanup

### PR DESCRIPTION
**Pull request by MSuih**
Mostly just consists of casting values explictly.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellSpursSpu.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ bool spursDma(spu_thread& spu, u32 cmd, u64 ea, u32 lsa, u32 size, u32 tag)
 
 	if (cmd == MFC_GETLLAR_CMD || cmd == MFC_PUTLLC_CMD || cmd == MFC_PUTLLUC_CMD)
 	{
-		const u32 rv = spu.get_ch_value(MFC_RdAtomicStat);
+		const u32 rv = ::narrow<u32>(spu.get_ch_value(MFC_RdAtomicStat));
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellSpursSpu.cpp]: This is very wrong and would trigger an assertation in some cases, use static_cast.

File: rpcs3/Emu/Cell/Modules/cellVdec.cpp [hunk 2]
```diff
@@ -617,7 +617,7 @@ error_code cellVdecGetPicture(u32 handle, vm::cptr<CellVdecPicFormat> format, vm
 
 	if (notify)
 	{
-		auto vdec_ppu = idm::get<named_thread<ppu_thread>>(vdec->ppu_tid);
+		auto vdec_ppu = idm::get<named_thread<ppu_thread>>(::narrow<u32>(vdec->ppu_tid));
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVdec.cpp]: statis_cast suffices, it never goes above UINT32_MAX.

File: rpcs3/Emu/Cell/Modules/cellVdec.cpp [hunk 3]
```diff
@@ -473,7 +473,7 @@ static error_code vdecOpen(ppu_thread& ppu, T type, U res, vm::cptr<CellVdecCb>
 	ppu_execute<&sys_ppu_thread_create>(ppu, +_tid, 0x10000, vid, +res->ppuThreadPriority, +res->ppuThreadStackSize, SYS_PPU_THREAD_CREATE_INTERRUPT, +_name);
 	*handle = vid;
 
-	const auto thrd = idm::get<named_thread<ppu_thread>>(*_tid);
+	const auto thrd = idm::get<named_thread<ppu_thread>>(::narrow<u32>(*_tid));
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVdec.cpp]: static_cast suffices.

File: rpcs3/cmake_modules/ConfigureCompiler.cmake [hunk 4]
```diff
@@ -9,6 +9,8 @@ if(MSVC)
 	#TODO: Some of these could be cleaned up
 	add_compile_options(/wd4805) # Comparing boolean and int
 	add_compile_options(/wd4804) # Using integer operators with booleans
+	add_compile_options(/wd4200) # Zero-sized array in struct/union
```
Comment (elad335) [inline rpcs3/cmake_modules/ConfigureCompiler.cmake]: wtf where have seen a zero sized array?

Comment (MSuih) [reply rpcs3/cmake_modules/ConfigureCompiler.cmake]: It's not being used in rpcs3 but libusb uses them a lot.

File: rpcs3/Emu/Cell/Modules/sceNp2.cpp [hunk 5]
```diff
@@ -188,7 +188,7 @@ error_code sceNp2Init(u64 poolsize, vm::ptr<void> poolptr)
 		return SCE_NP_ERROR_ALREADY_INITIALIZED;
 	}
 
-	const u32 result = std::bit_cast<u32>(sceNpInit(poolsize, poolptr));
+	const u32 result = std::bit_cast<u32>(sceNpInit(::narrow<u32>(poolsize), poolptr));
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/sceNp2.cpp]: It's fixed in another pr.

File: rpcs3/Emu/Cell/Modules/cellGcmSys.cpp [hunk 6]
```diff
@@ -425,7 +425,7 @@ s32 _cellGcmInitBody(ppu_thread& ppu, vm::pptr<CellGcmContextData> context, u32
 	vm::var<u64> _tid;
 	vm::var<char[]> _name = vm::make_str("_gcm_intr_thread");
 	ppu_execute<&sys_ppu_thread_create>(ppu, +_tid, 0x10000, 0, 1, 0x4000, SYS_PPU_THREAD_CREATE_INTERRUPT, +_name);
-	render->intr_thread = idm::get<named_thread<ppu_thread>>(*_tid);
+	render->intr_thread = idm::get<named_thread<ppu_thread>>(::narrow<u32>(*_tid));
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellGcmSys.cpp]: static_cast

File: rpcs3/Emu/System.cpp [hunk 7]
```diff
@@ -482,9 +482,9 @@ void Emulator::Init()
 						pdone = g_progr_pdone;
 
 						// Compute new progress in percents
-						const u32 total = ftotal + ptotal;
-						const u32 done = fdone + pdone;
-						const u32 new_value = double(done) * 100. / double(total ? total : 1);
+						const u32 total = ::narrow<u32>(ftotal + ptotal);
+						const u32 done = ::narrow<u32>(fdone + pdone);
```
Comment (elad335) [inline rpcs3/Emu/System.cpp]: I don't think the choice of using u64 for ftotal, fdone ...  g_progr_ftotal,  g_progr_fdone... over u32 was really wise here.
I'll let neko review this.

File: rpcs3/Emu/Cell/Modules/cellVdec.cpp [hunk 8]
```diff
@@ -617,7 +617,7 @@ error_code cellVdecGetPicture(u32 handle, vm::cptr<CellVdecPicFormat> format, vm
 
 	if (notify)
 	{
-		auto vdec_ppu = idm::get<named_thread<ppu_thread>>(vdec->ppu_tid);
+		auto vdec_ppu = idm::get<named_thread<ppu_thread>>(static_cast<u32>(vdec->ppu_tid));
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVdec.cpp]: ppu_tid actually needs to be declared as u32.

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 9]
```diff
@@ -1670,7 +1670,7 @@ void spu_thread::do_mfc(bool wait)
 	u32 fence = 0;
 
 	// Process enqueued commands
-	std::remove_if(mfc_queue + 0, mfc_queue + mfc_size, [&](spu_mfc_cmd& args)
+	static_cast<void>(std::remove_if(mfc_queue + 0, mfc_queue + mfc_size, [&](spu_mfc_cmd& args)
```
Comment (elad335) [inline rpcs3/Emu/Cell/SPUThread.cpp]:
```suggestion
	+std::remove_if(mfc_queue + 0, mfc_queue + mfc_size, [&](spu_mfc_cmd& args)
```

Comment (elad335) [reply rpcs3/Emu/Cell/SPUThread.cpp]: actually.
```suggestion
	0, std::remove_if(mfc_queue + 0, mfc_queue + mfc_size, [&](spu_mfc_cmd& args)
```
better marks the intent.

Comment (MSuih) [reply rpcs3/Emu/Cell/SPUThread.cpp]: I prefer casting to void, it's more clear what is going on

Comment (MSuih) [reply rpcs3/Emu/Cell/SPUThread.cpp]: It's also the recommended/standardized way of handling. For example [cppreference](https://en.cppreference.com/w/cpp/language/attributes/nodiscard) has "If, from a discarded-value expression **other than a cast to void,..**"

File: rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp [hunk 10]
```diff
@@ -391,7 +391,7 @@ error_code cellMsgDialogClose(f32 delay)
 	{
 		if (auto dlg = manager->get<rsx::overlays::message_dialog>())
 		{
-			thread_ctrl::spawn("cellMsgDialogClose() Thread", [=]
+			new named_thread("cellMsgDialogClose() Thread", [=]
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: Stop it. Stop silencing warnings you don't understand, you can't fix it like here.

**Pull request merged**